src/dialogs/more-info/controls/more-info-script.ts

```diff
@@ -138,17 +142,31 @@ class MoreInfoScript extends LitElement {
   protected override willUpdate(changedProperties: PropertyValues): void {
     super.willUpdate(changedProperties);
 
-    if (!changedProperties.has("stateObj")) {
-      return;
+    if (changedProperties.has("stateObj")) {
+      const oldState = changedProperties.get("stateObj") as
+        | HassEntity
+        | undefined;
+      const newState = this.stateObj;
+
+      if (
+        newState &&
+        (!oldState || oldState.entity_id !== newState.entity_id)
+      ) {
+        this._scriptData = {
+          action:
+            this.entry?.entity_id === newState.entity_id
+              ? this.entry.unique_id
```

Comment:
this is missing the domain name `script` now?

Reply:
Yeah, I think this branch of code is only exercised in a race condition which is rarely or maybe never hit, probably why I didn't notice it.